code/script.js

```diff
@@ -262,82 +273,135 @@ const selectQuestion = () => {
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
+        `Yes, the person wears ${attribute}! Keeping all the people that wears ${attribute}.`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
+        `No, the person doesn't wear ${attribute}! Removing all the people that wears ${attribute}.`
       )
     }
-  } else if (group === 'other') {
-    // Similar to the one above
+  } else if (category === 'hair color') {
+    if (keep) {
+    alert(`Yes, the person has ${value} hair! Keeping all the people with ${value} hair.`
+    )
   } else {
+    alert(
+      `No, the person doesn't have ${value} hair. Removing people with ${value} hair.`
+      )
+    }
+  } else if (category === 'eye color') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
-    } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+    alert(`Yes, the person has ${value} eyes! Keeping all the people with ${value} eyes.`
+    )
+  } else {
+    alert(`No, the person doesn't have ${value} eyes. Removing people with ${value} eyes.`
+      )
     }
+  } else if (category === 'other') {
+    if (keep) {
+      alert(`Yes, the person is a ${attribute}. Keeping all the people that are ${attribute}s.`
+      )
+  } else {
+      alert(`No, the person is not a ${attribute}. Removing all the people that are ${attribute}s.`)
+    } 
   }
 
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
 
+  // filter to keep or remove based on the keep variable.
+  if (keep) {
+    //user guessed right
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    //user guessed wrong
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+  generateBoard()
   // Invoke a function to redraw the board with the remaining people.
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// When clicking guess, the player confirms that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  confirm(`Are you sure you want to guess on ${suspect}?`) ? checkMyGuess(suspect) : {}
 }
 
+//Scoreboard  counter
+let currentScore = 0 
+const checkScore = (suspect) => {
+  suspect === secret.name ? currentScore++ : currentScore--
+}
+const scoreBoard = () => {
+  let playerScore = currentScore
+  return scoreBoardText.innerHTML = `Your score is: ${playerScore}`
+}
+
+
```

Comment:
I like that you added a score board! I know that you said yourself that it wasn't working correctly (giving negative scores and such). I saw it still does that in the game. Unforunately I can't really help you with that, I can't see what's causing it either...